sklearn/utils/tests/test_multiclass.py

```diff
@@ -268,20 +270,31 @@ def test_unique_labels_mixed_types():
         unique_labels([["1", "2"], [2, 3]])
 
 
-def test_is_multilabel():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype",
+    [(None, None, None)] + list(yield_namespace_device_dtype_combinations()),
+)
+def test_is_multilabel(array_namespace, device, dtype):
```

Comment:
I think it would make sense to have the original test and a new one that only deals with the Array API version. In the Array API test we should check that it returns the same results as the Numpy version.

Now we have quite a few `if` statements inside the test function for sparse, dense, array api etc which makes it a bit tricky to read (I find).